lib/OfxParser/Parser.php

```diff
@@ -174,4 +176,34 @@ private function convertSgmlToXml($sgml)
 
         return trim($xml);
     }
+
+    /**
+     * Check for self closing tags
+     *
+     * @param string $xml
+     * @return string
+     */
+    private function closeSelfTags($xml)
```

Comment:
This should already be handled in `closeUnclosedXmlTags()` (called in `convertSgmlToXml`), since unclosed tags are a characteristic of OFX files... I'm wondering if perhaps there's another issue causing that not to pick up your closed tags - do you have a specific example of which tags in your sample OFX (`ofxdata-selfclose.ofx`) that isn't currently being fixed?

Reply:
I checked this problem when I imported my ofx bank file that came with the tag **BALAMT** without information. I saw in the code that `closeUnclosedXmlTags()` matches only cases like `<SOMETHING>blah`. That method is the place where I have to close unclosed tags?

Reply:
Ah I see now, yes this is a subtly different case. I'm not entirely sure off my head whether this would be a valid OFX file?

Reply:
Well, as my bank is an official company, I thought the file was correct...

Reply:
Ya, unfortuantely every financial institution seems to do things slightly differently :/ the ledger details are blank, which doesn't really make sense:

```
<LEDGERBAL>
<BALAMT>
<HUGO>
<DTASOF>20190924100000[-03:EST]
</LEDGERBAL>
```

If there is no balance to be shown, I'd either set the `BALAMT` to be `0.00` for example, or omit the `LEDGERBAL` tag entirely... :thinking:

Reply:
You're entirely right, but the tag HUGO was inserted for me, just to test 😅